### ISSUE TITLE
Create Iterator for iterating over vertices in degree order

### ISSUE THREAD

**Issue by LucaStaus** [sipura/Sipura#67]
We should add Iterators to the Traversal object that can iterate over all vertices in non decreasing and non increasing degree order.

**Comment by moritz-gross** [comment]
If we just have an iterator for the vertices, we would have to use .map{g.degreeOf(it)} or something similar every time we want to have the degree sequence.

Do we want to add both vertex- & degree-iterator or just one of them? @LucaStaus

**Comment by LucaStaus** [comment]
I dont think we need an extra method for that. I think it is fine to just have the iterator over the vertices since degreeOf has very little overhead.
